replicaset.go

```diff
@@ -313,13 +341,25 @@ type mgoSession interface {
 	Close()
 	SetMode(consistency mgo.Mode, refresh bool)
 	DB(name string) *mgo.Database
+	BuildInfo() (mgo.BuildInfo, error)
 }
 
+var repairNeeded = errors.ConstError("repair needed")
+
 // applyReplSetConfig applies the specified config to the mongo session.
 // It checks if the replica set changes cause the DB connection to be dropped.
 // If so, it Refreshes the session and tries to Ping again.
 func applyReplSetConfig(cmd string, session mgoSession, newConfig Config) error {
 	err := session.Run(bson.D{{"replSetReconfig", newConfig}}, nil)
+	if err != nil {
+		qe := err.(*mgo.QueryError)
```

Comment:
this looks dangerous

Reply:
`Run` only ever returns `QueryError` but I can add a guard just in case that changes upstream